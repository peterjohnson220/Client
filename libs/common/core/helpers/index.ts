export * from './kendo-grid-filter.helper';
