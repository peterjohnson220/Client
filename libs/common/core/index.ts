export * from './pipes';
export * from './helpers';
