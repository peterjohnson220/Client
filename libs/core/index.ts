export * from './helpers';
export * from './pipes';
export * from './common.module';
