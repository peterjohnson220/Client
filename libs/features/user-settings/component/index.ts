export * from './auto-share-modal';
export * from './auto-shared-users';
export * from './shareable-user';
