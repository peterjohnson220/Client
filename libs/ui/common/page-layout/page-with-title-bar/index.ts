export * from './page-with-title-bar.component';
