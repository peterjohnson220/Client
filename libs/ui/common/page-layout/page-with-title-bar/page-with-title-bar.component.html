<div class="flex-column-container">
  <div class="page-header d-flex justify-content-between align-items-center">

    <div class="d-flex">
      <button *ngIf="returnUrl"
              class="btn btn-secondary back-btn"
              [routerLink]="returnUrl">
        <i class="fas fa-long-arrow-alt-left" aria-hidden="true"></i>
      </button>
      <a href="{{absoluteUrl}}"
         *ngIf="absoluteUrl"><button class="btn btn-secondary back-btn">
        <i class="fas fa-long-arrow-alt-left" aria-hidden="true"></i>
      </button> </a>

      <h3><ng-content select="[page-title]"></ng-content></h3>
    </div>

    <div class="d-flex space-out-children">
      <ng-content select="[page-actions]"></ng-content>
    </div>

  </div>

  <div class="page-content" [class.p-0]="contentNoPadding">
    <ng-content select="[page-content]"></ng-content>
  </div>
</div>
