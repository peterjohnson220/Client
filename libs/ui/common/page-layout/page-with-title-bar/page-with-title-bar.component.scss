@import 'colors';

.page-header{
  background: $grey;
  border-bottom: 1px solid $border-grey;
  height: 67px;
  flex-shrink: 0;
  padding: 0 0.85rem;
}

.page-header h3{
  margin-bottom: 3px;
}

.back-btn{
  padding: 0.375rem 0.95rem;
  margin-right: 0.65rem;
}

:host ::ng-deep .space-out-children > *{
  margin-left: 0.5rem;
}
