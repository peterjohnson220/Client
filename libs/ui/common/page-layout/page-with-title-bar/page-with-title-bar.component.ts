import { Component, Input } from '@angular/core';
import { Router } from '@angular/router';

import { RouteTrackingService } from 'libs/core/services';

@Component({
  selector: 'pf-page-with-title-bar',
  templateUrl: './page-with-title-bar.component.html',
  styleUrls: ['./page-with-title-bar.component.scss']
})
export class PageWithTitleBarComponent {
  @Input() isFullHeader = true;
  @Input() returnUrl: string;
  @Input() goBackToPreviousRoute: boolean;
  @Input() absoluteUrl: string;
  @Input() contentNoPadding: boolean;

  constructor(
    private routeTrackingService: RouteTrackingService,
    private router: Router
  ) {}

  goBack() {
    this.goBackToPreviousRoute ? this.routeTrackingService.goBack() : this.router.navigate([this.returnUrl]);
  }
}
