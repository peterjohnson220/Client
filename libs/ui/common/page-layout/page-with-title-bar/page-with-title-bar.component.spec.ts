import { Component, NO_ERRORS_SCHEMA } from '@angular/core';
import { TestBed, ComponentFixture } from '@angular/core/testing';
import { By } from '@angular/platform-browser';

import { PageWithTitleBarComponent } from './page-with-title-bar.component';

// Host Component for testing transclusion
@Component({
  template: `
    <pf-page-with-title-bar>
      <ng-container page-title>I am a page title</ng-container>
      <ng-container page-actions><button>I am a page action</button></ng-container>
      <ng-container page-content><h1>I am the page content</h1></ng-container>
    </pf-page-with-title-bar>`
})
class TestHostComponent {}


describe('Page With Title Bar', () => {
  let fixture: ComponentFixture<PageWithTitleBarComponent>;
  let instance: PageWithTitleBarComponent;

  // Configure Testing Module for before each test
  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [
        PageWithTitleBarComponent, TestHostComponent
      ],
      // Shallow Testing
      schemas: [ NO_ERRORS_SCHEMA ]
    });

    fixture = TestBed.createComponent(PageWithTitleBarComponent);
    instance = fixture.componentInstance;
  });

  it('should show the back button, when provided a return Url', () => {

    instance.returnUrl = '/returnToThis/Path';

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should transclude into slots', () => {

    const hostComponent = TestBed.createComponent(TestHostComponent);

    expect(hostComponent).toMatchSnapshot();
  });

});
