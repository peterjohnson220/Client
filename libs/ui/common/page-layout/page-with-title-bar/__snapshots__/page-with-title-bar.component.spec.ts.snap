// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page With Title Bar should show the back button, when provided a return Url 1`] = `
<pf-page-with-title-bar
  returnUrl={[Function String]}
>
  <div
    class="flex-column-container"
  >
    
    
    <div
      class="page-header d-flex justify-content-between align-items-center"
    >
      
  
      
      <div
        class="d-flex"
      >
        
        
        
        <button
          class="btn btn-secondary back-btn"
        >
          
          
          <i
            aria-hidden="true"
            class="fas fa-long-arrow-alt-left"
          />
          
        
        </button>
        
  
        
        <h3 />
        
      
      </div>
      
  
      
      <div
        class="d-flex space-out-children"
      >
        
        
        
      
      </div>
      
  
    
    </div>
    
  
    
    <div
      class="page-content"
    >
      
      
      
    
    </div>
    
  
  </div>
  
</pf-page-with-title-bar>
`;

exports[`Page With Title Bar should transclude into slots 1`] = `
<ng-component>
  
      <pf-page-with-title-bar>
    <div
      class="flex-column-container"
    >
      
    
      <div
        class="page-header d-flex justify-content-between align-items-center"
      >
        
  
      
        <div
          class="d-flex"
        >
          
        
          
          
  
        
          <h3>
            
            I am a page title
          </h3>
          
      
        </div>
        
  
      
        <div
          class="d-flex space-out-children"
        >
          
        
          
          <button>
            I am a page action
          </button>
          
      
        </div>
        
  
    
      </div>
      
  
    
      <div
        class="page-content"
      >
        
      
        
        <h1>
          I am the page content
        </h1>
        
    
      </div>
      
  
    </div>
    
  
  </pf-page-with-title-bar>
</ng-component>
`;
